network/framework/src/peer/mod.rs

```diff
@@ -330,30 +331,44 @@ where
         mut writer: MultiplexMessageSink<impl AsyncWrite + Unpin + Send + 'static>,
         max_frame_size: usize,
         max_message_size: usize,
-    ) -> (aptos_channels::Sender<NetworkMessage>, oneshot::Sender<()>) {
+    ) -> (
+        aptos_channel::Sender<(), NetworkMessage>,
+        oneshot::Sender<()>,
+    ) {
         let remote_peer_id = connection_metadata.remote_peer_id;
-        let (write_reqs_tx, mut write_reqs_rx): (aptos_channels::Sender<NetworkMessage>, _) =
-            aptos_channels::new(1024, &counters::PENDING_WIRE_MESSAGES);
+        let (write_reqs_tx, mut write_reqs_rx): (aptos_channel::Sender<(), NetworkMessage>, _) =
+            aptos_channel::new(
+                QueueStyle::KLAST,
+                1024,
+                Some(&counters::PENDING_WIRE_MESSAGES),
```

Comment:
We should make this guage a vec guage and take in by peer id as label so we can distinguish per peer writer.

Reply:
I thought we can't do the n^2 time series

Reply:
I think it's okay. our metrics backend can support. i think we already do in a bunch of places. Otherwise, this gauge isn't very useful.